### PR TITLE
[ODS-3670] Remove composites API "hierarchical" feature

### PR DESCRIPTION
**Pull request by gmcelhanon**
Paired with https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/pull/82.

### CODE REVIEW

File: Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs [hunk 1]
```diff
@@ -84,95 +84,91 @@ private IEnumerable<object> GetClasses(Aggregate aggregate, Entity entity)
...
-                                                               AggregateRelativeNamespace = a.OtherEntity.GetRelativeAggregateNamespace(
-                                                                   TemplateContext.SchemaProperCaseName,
-                                                                   isQueryModel: true,
-                                                                   isExtensionContext: TemplateContext.IsExtension),
-                                                               ClassName = a.OtherEntity.Name, ClassNameSuffix = QueryModelSuffix,
-                                                               AssociationName = a.Name
-                                                           })
-                        };
-                }
+                yield return
```
Comment (jamessmckay) [inline Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs:87]: can you please fix the formatting on this class while you are at it.

File: Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature [hunk 2]
```diff
@@ -1,52 +0,0 @@
-﻿@Composites @HierarchicalComposite @Ignore
```
Comment (jamessmckay) [inline Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature:1]: Is there a any unused code in the composite steps that can be deleted?

File: Application/EdFi.Ods.Standard/Artifacts/Metadata/DatabaseViews.generated.json [hunk 3]
```diff
@@ -923,504 +923,6 @@
       }
     ]
   },
-  {
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Standard/Artifacts/Metadata/DatabaseViews.generated.json:926]: was this generated or manually edited?

File: Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs [hunk 4]
```diff
@@ -251,18 +238,10 @@ public void ApplyRootResource(CompositeDefinitionProcessorContext processorConte
...
             builderContext.CurrentAlias = builderContext.AliasGenerator.GetNextAlias();
 
-            bool isHierarchicalResource =
-                processorContext.ShouldUseHierarchy()
-                && processorContext.CurrentResourceClass.Entity.HasSelfReferencingAssociations
-                || processorContext.ShouldUseReferenceHierarchy();
-
             builderContext.From.AppendFormat(
-                "\r\n\tjoin {0}.{1}{2} {3}",
+                "\r\n\tjoin {0}.{1} {2}",
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs:242]: should you use Environment.NewLine for the \r\n instead to make this more friendly for other operating systems?

File: Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs [hunk 5]
```diff
@@ -404,29 +383,8 @@ public HqlBuilderContext CreateFlattenedMemberContext(HqlBuilderContext builderC
...
-                   .Intersect(safelySortableProperties);
-            }
-            else
-            {
-                orderByProperties = locallyDefinedIdentifyingProperties;
-            }
-
             // Add ORDER BY for the primary keys
-            orderByProperties.ForEach(
+            locallyDefinedIdentifyingProperties.ForEach(
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs:387]: Is there a change for the approval tests for composites? If so did you deleted the saved file?

File: Application/EdFi.Ods.Common/Composites/CompositeDefinitionProcessorContext.cs [hunk 6]
```diff
@@ -66,10 +66,6 @@ public class CompositeDefinitionProcessorContext
 
         public bool ShouldIncludeResourceSubtype() => AttributeValueAsBool(CompositeDefinitionHelper.IncludeResourceSubtype);
 
-        public bool ShouldUseHierarchy() => AttributeValueAsBool(CompositeDefinitionHelper.UseHierarchy);
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Composites/CompositeDefinitionProcessorContext.cs:69]: Is there a custom attribute that needs to be removed?

File: Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd [hunk 7]
```diff
@@ -65,7 +65,7 @@
       <xs:element name="Property" type="Property" />
       <xs:element name="Collection" type="Collection" />
       <xs:element name="ReferencedResource" type="ReferencedResource" />
-      <xs:element name="LinkedCollection" type="LinkedCollection" />
+      <xs:element name="LinkedCollection" type="Resource" />
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd:68]: Why are we changing to a resource?

File: Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd [hunk 8]
```diff
@@ -108,19 +106,11 @@
       <xs:element name="Collection" type="Collection" />
       <xs:element name="ReferencedResource" type="ReferencedResource" />
       <xs:element name="EmbeddedObject" type="EmbeddedObject" />
-      <xs:element name="LinkedCollection" type="LinkedCollection" />
+      <xs:element name="LinkedCollection" type="Resource" />
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd:109]: Why the change to resource?

File: Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs (hunk 1 above)
Comment (gmcelhanon) [reply Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs:87]: You're good with the other changes to this file?

File: Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature (hunk 2 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature:1]: I looked. There are precious few steps used by the feature, and I could not find any steps that supported the language. So, I don't believe so.

File: Application/EdFi.Ods.Standard/Artifacts/Metadata/DatabaseViews.generated.json (hunk 3 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Standard/Artifacts/Metadata/DatabaseViews.generated.json:926]: I definitely did not modify the `.generated.json` file manually. I went through the whole CI packaging and initdev process dance to achieve these results.

File: Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs (hunk 4 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs:242]: Should we create a ticket to replace all of these globally, rather than take them on in parts and pieces? Seems it would be more efficient to just attack them all at once in a separate ticket.

File: Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs (hunk 1 above)
Comment (jamessmckay) [reply Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs:87]: yes, just need to reformat the file

File: Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature (hunk 2 above)
Comment (jamessmckay) [reply Application/EdFi.Ods.WebService.Tests/Composites/Composites support hierarchical resources.feature:1]: ok

File: Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs (hunk 5 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs:387]: This was a refactoring where code was removed that was no longer needed, but functionality and output should not have been affected for the non-hierarchical use cases. In re-reviewing the code, I see no approval tests for the _hierarchical_ SpecFlow feature.

File: Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd (hunk 7 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd:68]: Because the _only_ reason for the `LinkedCollection` type was to add support for `useHierarchy` attribute on `LinkedCollection` (which is based on `Resource` type) and that attribute is no longer supported.

File: Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd (hunk 8 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Metadata/Schemas/Ed-Fi-ODS-API-Composites.xsd:109]: See above.

File: Application/EdFi.Ods.Common/Composites/CompositeDefinitionProcessorContext.cs (hunk 6 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Composites/CompositeDefinitionProcessorContext.cs:69]: As in a .NET custom attribute? No.

For reference:
```private bool AttributeValueAsBool(string attributeName)
{
    bool.TryParse(AttributeValue(attributeName), out bool result);
    return result;
}
```
... and ...

```
public string AttributeValue(string attributeName) => CurrentElement.AttributeValue(attributeName);
```

This code is extracting the attribute that was removed from the composites definition XML schema.

File: Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs (hunk 4 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/NHibernate/Composites/HqlBuilder.cs:242]: See https://tracker.ed-fi.org/browse/ODS-4410.

File: Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs (hunk 1 above)
Comment (gmcelhanon) [reply Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/EntitiesForQueries.cs]: Done.
